esmvaltool/diag_scripts/hydrology/globwat.py

```diff
@@ -0,0 +1,340 @@
+                                            group_metadata,
+                                            run_diagnostic)
+
+
+logger = logging.getLogger(Path(__file__).name)
+
+
+def create_provenance_record():
+    """Create a provenance record."""
+    record = {
+        'caption': "Forcings for the GlobWat hydrological model.",
+        'domains': ['global'],
+        'authors': [
+            'abdollahi_banafsheh',
+            'alidoost_sarah',
+        ],
+        'projects': [
+            'ewatercycle',
+        ],
+        'references': [
+            'acknow_project',
+        ],
+        'ancestors': [],
+    }
+    return record
+
+
+def rechunk_and_regrid(src, tgt, scheme):
+    """Rechunk cube src and regrid it onto the grid of cube tgt."""
+    src_chunks = compute_chunks(src, tgt)
+    src.data = src.lazy_data().rechunk(src_chunks)
+    return regrid(src, tgt, scheme)
+
+
+def change_data_type(cube):
+    """Change data type to float32."""
+    cube.data = cube.core_data().astype('float32')
+    for coord_name in 'latitude', 'longitude', 'time':
+        coord = cube.coord(coord_name)
+        coord.points = coord.core_points().astype('float32')
+        coord.bounds = None
+        coord.guess_bounds()
+    return cube
+
+
+def _convert_units(cube):
+    """Convert unit of cube.
+
+    From kg m-2 s-1 to kg m-2 month-1 or kg m-2 day-1.
+    Note that the unit kg m-2 s-1 is equivalent to mm s-1.
+    """
+    mip = cube.attributes['mip']
+
+    if mip == 'Amon':
+        cube.convert_units('kg m-2 month-1')  # equivalent to mm/month
+    elif mip == 'day':
+        cube.convert_units('kg m-2 day-1')  # equivalent to mm/day
+    return cube
+
+
+def _fix_negative_values(cube):
+    """Change negative values to zero."""
+    cube.data = da.where(cube.core_data() < 0, 0, cube.core_data())
+    return cube
+
+
+def get_input_cubes(metadata):
+    """Return a dictionary with all (preprocessed) input files."""
+    provenance = create_provenance_record()
+    all_vars = {}
+    for attributes in metadata:
+        short_name = attributes['short_name']
+        filename = attributes['filename']
+        logger.info("Loading variable %s", short_name)
+        cube = iris.load_cube(filename)
+        all_vars[short_name] = change_data_type(cube)
+        cube.attributes['mip'] = attributes['mip']
+        provenance['ancestors'].append(filename)
+    return all_vars, provenance
+
+
+def load_target(cfg):
+    """Load target grid."""
+    filename = Path(cfg['auxiliary_data_dir']) / cfg['target_file']
+    cube = iris.load_cube(str(filename))
+    return cube
+
+
+def langbein_pet(tas):
+    """Calculate potential ET using Langbein method.
+
+    Langbein curve is a temperature-based method for calculating potential ET.
+    In this equation, T is the annual average temperature in degree Celcius,
+    and pet is in mm per year.
+    Reference: https://doi.org/10.3133/cir52
+    """
```

Comment:
Reading the publication it is a bit difficult to extract the formulation of the method. The usage of constant A, B, C does not help in that respect. Could you add comments as to what exactly these contants represent?

Reply:
More explanation added: The Langbein curve represents an empirical relationship between temperature and potential ET (pet). Where T is the annual average temperature in degree Celcius, pet is in mm per year, and a, b, c are unitless empirical constants.
Reference: https://doi.org/10.3133/cir52 page 8, figure 1.
An example of using Langbein method can be found at:
https://doi.org/10.1080/02626667.2017.1332416 page 1472, equation 7.